### PR TITLE
WIP: BXMSDOC-7090 Fail-over of a RUNNING job in kie-server executor

### PR DESCRIPTION
**Pull request by emmurphy1**
[RHPAM](http://file.ork.redhat.com/~emmurphy/BXMSDOC-7090-pam/#clustering-kie-server-failover-proc_clustering-runtime-standalone)
[RHDM](http://file.ork.redhat.com/~emmurphy/BXMSDOC-7090-dm/#clustering-kie-server-failover-proc_clustering-runtime-standalone)

[BXMSDOC-7090](https://issues.redhat.com/browse/BXMSDOC-7090)

### CODE REVIEW

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,106 @@
+[id='clustering-kie-server-failover-proc_{context}']
+= Failover for jobs in the RUNNING state on failed {KIE_SERVER} nodes
```
Comment (michelehaglund) [inline doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Needs to be changed to a procedure description. How about:
Configuring failover for jobs in the RUNNING state on failed {KIE_SERVER} nodes

If that title is too long, you might simple say "Configuring failover"

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+By default, when multiple {KIE_SERVER} nodes are running in cluster, if one {KIE_SERVER} node fails, other available {KIE_SERVER} nodes pick up any jobs in the `QUEUED` or `RETRYING` state from the failed {KIE_SERVER} node. However, these nodes do not pick up jobs that are in the `RUNNING` state.
+
+You can configure your cluster to failover all jobs when a {KIE_SERVER} node fails, including jobs in the `RUNNING` state, to be requeued to another {KIE_SERVER} node. If there is timer start node in a process on the failed {KIE_SERVER} node, the process is started on only one of the nodes.
```
Comment (michelehaglund) [inline doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: "You can configure your cluster to failover all jobs when a {KIE_SERVER} node fails, including jobs in the `RUNNING` state, to be requeued to another {KIE_SERVER} node. If there is timer start node in a process on the failed {KIE_SERVER} node, the process is started on only one of the nodes."

Possible rewrite:
You can configure your cluster to include jobs in the `RUNNING` state, to be requeued to another {KIE_SERVER} node. If there is timer start node in a process on the failed {KIE_SERVER} node, the process is started on only one of the nodes.

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+<socket-binding name="jgroups-mping" interface="private" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45700"/>
+
+  <socket-binding name="jgroups-tcp" interface="private" port="7600"/>
+  <socket-binding name="jgroups-tcp-fd" interface="private" port="57600"/>
+  <socket-binding name="jgroups-udp" interface="private" port="55200" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45688"/>
+  <socket-binding name="jgroups-udp-fd" interface="private" port="54200"/>
+----
+. Install the Infinispan extension:
+.. Open the `kie-server.war/WEB-INF/jboss-deployment-structure.xml` file in a text editor and add the `jboss-deployment-structure.xml` below the following dependencies:
```
Comment (michelehaglund) [inline doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Can you add an example of how the jboss-deployment-structure.xml will look?

**Review by michelehaglund — APPROVED** [review]
Looks good @emmurphy1 Comments added.

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc (hunk 1 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Wow, great catch!

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc (hunk 3 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Antonio,  can you help with an example?

**Review by afalhambra — CHANGES_REQUESTED** [review]
@emmurphy1 Overall looks pretty good, just some comments and suggestions. One thing I would also highlight is that this configuration is only valid for EAP server, nothing else. 
Let me know what you think.

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc [hunk 4]
```diff
@@ -0,0 +1,106 @@
+[id='clustering-kie-server-failover-proc_{context}']
+= Configuring failover for jobs in the RUNNING state on failed {KIE_SERVER} nodes
+
+By default, when multiple {KIE_SERVER} nodes are running in cluster, if one {KIE_SERVER} node fails, other available {KIE_SERVER} nodes pick up any jobs in the `QUEUED` or `RETRYING` state from the failed {KIE_SERVER} node. However, these nodes do not pick up jobs that are in the `RUNNING` state.
+
+You can configure your cluster to include jobs in the RUNNING state, to be requeued to another {KIE_SERVER} node. If there is timer start node in a process on the failed {KIE_SERVER} node, the process is started on only one of the nodes.
```
[the remark this answers is not in the document]
Comment (emmurphy1) [reply doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: @afalhambra I think that @elguardian  wants to call out the information in that sentence.

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc (hunk 3 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Thank you @afalhambra . I don't think we need to include the example in the doc.

File: doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (afalhambra) [reply doc-content/enterprise-only/installation/clustering-kie-server-failover-proc.adoc]: Ok then.

**Review by afalhambra — APPROVED** [review]
Great job @emmurphy1 - I think you've addressed all my comments. Maybe we could highlight that EAP server restriction, but maybe it's enough with the `prerequisites` section.
Anyways it's fine with me

**Pull request merged**